config/application.rb

```diff
@@ -11,30 +11,35 @@
 # you've limited to :test, :development, or :production.
 Bundler.require(*Rails.groups)
 
-$rubygems_config = YAML.load_file("config/rubygems.yml")[Rails.env].symbolize_keys
-HOST             = $rubygems_config[:host]
-
 module Gemcutter
   class Application < Rails::Application
+    def config_for(name, env = Rails.env)
+      YAML.load_file(Rails.root.join("config/#{name}.yml"))[env]
+    end
+
     config.time_zone = "UTC"
     config.encoding  = "utf-8"
 
     config.middleware.use "Hostess"
-    config.middleware.insert_after "Hostess", "Redirector" if $rubygems_config[:redirector] && ENV["LOCAL"].nil?
+    if Application.config_for(:rubygems)['redirector'] && ENV["LOCAL"].nil?
+      config.middleware.insert_after "Hostess", "Redirector"
+    end
 
     unless Rails.env.maintenance?
-      config.action_mailer.default_url_options  = { :host => HOST }
-      config.action_mailer.delivery_method      = $rubygems_config[:delivery_method]
+      config.action_mailer.default_url_options  = { host: Application.config_for(:rubygems)['host'] }
```

Comment:
If you could do that, would be great, also do it in a separate commit, so that change is isolated. thanks